patches/minecraft/net/minecraft/world/storage/MapDecoration.java.patch

```diff
@@ -0,0 +1,32 @@
+--- ../src-base/minecraft/net/minecraft/world/storage/MapDecoration.java
++++ ../src-work/minecraft/net/minecraft/world/storage/MapDecoration.java
+@@ -1,6 +1,8 @@
+ package net.minecraft.world.storage;
+ 
+ import net.minecraft.util.math.MathHelper;
++import net.minecraftforge.fml.common.FMLCommonHandler;
++import net.minecraftforge.fml.common.FMLLog;
+ import net.minecraftforge.fml.relauncher.Side;
+ import net.minecraftforge.fml.relauncher.SideOnly;
+ 
+@@ -121,6 +123,13 @@
+             this.field_191177_m = p_i47344_4_;
+         }
+ 
++        private Type(byte icon, boolean renderedOnFrame, int mapColor)
```

Comment:
What's wrong with the old one?

Reply:
We can reassign the variable icon (final). We need to create a new constructeur and use it.

Reply:
For this one, you need to edit enum helper test to ignore the old constructor.

Reply:
How can I do that properly?